[DX-1655 TT-12355] added docs on gateway log format (#5346)
TykTechnologies/tyk-docs: tyk-docs/content/log-data.md

---
date: 2017-03-24T12:53:50Z
title: Logging system and API events

> tags: ["logging", "observability", "system events", "logs", "json", "log format"]

Reviewer: @LLe27 The tags are used for search indexing and headings. Maybe also add tags for JSON, logging, logging format. In a separate comment I have made suggestions for sub headings for Standard and JSON format logging

If unset or left empty, it will default to `info`. 

### Setting log format (only available for the Gateway)

> As of Tyk Gateway versions LTS `v5.3` and `v5.6.0`, you can control log format using the `TYK_LOGFORMAT` environment variable. By default, logs are in standard format, but setting `TYK_LOGFORMAT` to `json` will output logs in JSON format.

Reviewer 2: ```suggestion
As of Tyk Gateway versions LTS `v5.3` and `v5.6.0`, you can control log format using the `TYK_LOGFORMAT` environment variable. By default, logs are in `default` format, but setting `TYK_LOGFORMAT` to `json` will output logs in JSON format.
```
This environment variable accepts only two options: `default` or `json`. Let's refer the standard format as the default format.

Reference Code: https://github.com/TykTechnologies/tyk/blob/master/log/log.go#L71

Author: @sharadregoti correct! Thanks for the review.

> **Standard Format:**

Reviewer: ```suggestion
##### Standard logging format
```

> ##### Standard logging format

Reviewer 2: ```suggestion
##### Default logging format
```

```
time="Sep 05 09:04:12" level=info msg="Tyk API Gateway v5.6.0" prefix=main
```

> **JSON Format:**

Reviewer: ```suggestion
##### JSON logging format
```

```
{"level":"info","msg":"Tyk API Gateway v5.6.0","prefix":"main","time":"2024-09-05T09:01:23-04:00"}
```

> 

Reviewer 3: ```suggestion
As a general performance tuning note, the `json` output provides less allocation overhead than the standard logger. When tuning for performance, it's recommended to use the JSON logging format.
```

> As a general performance tuning note, the `json` output provides less allocation overhead than the standard logger. When tuning for performance, it's recommended to configure the JSON logging format.

Reviewer 2: ```suggestion
{{< note >}}
**Note**  
As a general performance tip, the `json` output format incurs less memory allocation overhead than the default logger. For optimal performance, it's recommended to configure logging in the JSON format.
{{< /note >}}
```

@LLe27 When you are referring to less allocation, it is with respect to `memory` right?

If unset or left empty, it will default to `info`. 

### Setting log format (only available for the Gateway)

> As of Tyk Gateway version `v5.6.0`, you can control log format using the `TYK_GW_LOGFORMAT` environment variable. By default, logs are in standard format, but setting `TYK_GW_LOGFORMAT` to `json` will output logs in JSON format.

Reviewer 3: This needs adjustment to list the 5.3 LTS release as well

Reviewer 3: Document as above, TYK_LOGFORMAt / TYK_LOGLEVEL (`_GW` prefix isn't in use in this doc)

Author: Gotcha I wasn't sure cause I only saw `v5.6.0` listed in the fixed versions. Just to confirm both LTS `v5.3` and `v5.6.x` will include these changes?

Author: Will update the configs accordingly

Reviewer 3: @LLe27 for now, how about opening a dashboard PR? :)

Author: @titpetric that's on my todo list, with your guidance of course! ;)

If unset or left empty, it will default to `info`. 

### Setting log format (only available for the Gateway)

> As of Tyk Gateway versions LTS `v5.3` and `v5.6.0`, you can control log format using the `TYK_LOGFORMAT` environment variable. By default, logs are in `default` format, but setting `TYK_LOGFORMAT` to `json` will output logs in JSON format.

Author: I think it's slotted for `v5.6.0` now. @titpetric can you confirm which version of `v5.3.x` it is merged to? Thankjs

> As of Tyk Gateway versions LTS `v5.3` and `v5.6.0`, you can control log format using the `TYK_LOGFORMAT` environment variable. By default, logs are in `default` format, but setting `TYK_LOGFORMAT` to `json` will output logs in JSON format.

Reviewer 4: we'll put 5.3.6 based on the changelog of 5.3.6 FYI @andyo-tyk @lghiur

##### Default logging format
```
time="Sep 05 09:04:12" level=info msg="Tyk API Gateway v5.6.0" prefix=main
```

##### JSON logging format

> ```

Reviewer 4: ```suggestion
```json
```
